src/rgw/rgw_sync_policy.h

```diff
@@ -675,6 +675,16 @@ struct rgw_sync_policy_info {
     return groups.empty();
   }
 
+  bool is_directional() const {
+    for (auto& item : groups) {
+      auto& group = item.second;
+      if (!group.data_flow.directional.empty()) {
```

Comment:
ok, so the presence of a directional flow would mean that we don't have symmetry within the zonegroup. but does `!is_directional()` guarantee symmetry, or is it possible that the group doesn't include all of the zonegroup's zones?

Reply:
yes it is possible that the groups may not contain all zones, which probably means that a zone opted out of sync and it can have its own data. think object prefix filtering also needs to be checked.

Reply:
b3e05c3531ca289f6d90326ee3bf0f6a9fa43adc covers two more cases: zones excluded from sync and object filtering.

Reply:
@cbodley teuthology runs with sync policy test cases haven't been passing consistently. but they pass when I test locally. can you take a look at https://github.com/ceph/ceph/commit/b3e05c3531ca289f6d90326ee3bf0f6a9fa43adc please?